src/main/java/space/obminyashka/items_exchange/controller/UserController.java

```diff
@@ -219,4 +220,14 @@ private User getUser(String username) {
         return userService.findByUsernameOrEmail(username).orElseThrow(() -> new UsernameNotFoundException(
                 getMessageSource(ResponseMessagesHandler.ExceptionMessage.USER_NOT_FOUND)));
     }
+
+    private void validateEmail(User user, String email) throws DataConflictException {
```

Comment:
The general approach - fewer privileges to all the data passing to any method.
If you check the content of the method the **user** is used to get it's email only, so it would be much reasonable to pass only 2 emails: currentEmail and newEmail